### PR TITLE
Added tests for DT[, .SD] retaining secondary indices, #1709

### LINKED ISSUE #1709: DT[, .SD] doesn't seem to retain secondary indices
``` r
require(data.table)
dt = data.table(x=1:5, y=6:10)
setindex(dt, x)

indices(dt) # [1] "x"
indices(dt[, .SD]) # NULL
```

### PR DESCRIPTION
**Pull request by joshhwuu**
Closes #1709 

Added tests to ensures that setting an index for a datatable with `setindex()` correctly retains secondary indicies.

### CODE REVIEW

**Review by jangorecki — APPROVED** [review]

**Comment by ben-schwen** [comment]
The tests look good. Ideally, you would git bisect the repo to check where the initial fix for #1709 was made and append the tests to the tests of that PR.

**Comment by joshhwuu** [comment]
> The tests look good. Ideally, you would git bisect the repo to check where the initial fix for #1709 was made and append the tests to the tests of that PR.

Thanks for the suggestion! I'm not sure where the fix for #1709 is, I just saw:

> ```r
> require(data.table)
> dt = data.table(x=1:5, y=6:10)
> setindex(dt, x)
> 
> indices(dt)
> #[1] "x"
> indices(dt[, .SD])
> #[1] "x"
> ```
> 
> it now does retain index. we can close this issue by adding a unit test

I'll try to dig around and see if I find the PR that fixed this issue!

**Comment by MichaelChirico** [comment]
Hmm, just to find a baseline of when this wasn't working I tried 1.13.0 (2020), it was already working then. But I can't seem to install any earlier versions, getting linker failures e.g.

```
/usr/bin/ld: inrange.o:/home/michael/github/data.table/src/data.table.h:80: multiple definition of `SelfRefSymbol'; init.o:/home/michael/github/data.table/src/init.c:30: first defined here
...
```

Same issue on all of 1.11.0, 1.12.0, 1.12.8, at least when building from source. So `git bisect` may be difficult here. I'll just go ahead and merge so this PR doesn't spiral out in difficulty, if anyone wants to dig further please feel free & we can add a commit later.

File: inst/tests/tests.Rraw [hunk 1]
```diff
@@ -18350,3 +18350,12 @@ if (test_bit64) {
...
 }
+
+# Unit tests for DT[, .SD] retaining secondary indices, #1709
+DT = data.table(x=1:5, y=6:10)
+setindex(DT, x)
+test(2249.1, indices(DT), 'x')
+test(2249.2, indices(DT[, .SD]), 'x')
+setindex(DT, y)
+test(2249.3, indices(DT), c('x', 'y'))
+test(2249.4, indices(DT[, .SD]), c('x', 'y'))
```
Comment (MichaelChirico) [inline inst/tests/tests.Rraw:18361]:
```suggestion
test(2249.4, indices(DT[, .SD]), c('x', 'y'))

```

**Review by MichaelChirico — APPROVED** [review]

**Comment by MichaelChirico** [comment]
Thanks!

**Pull request merged**